### PR TITLE
[TFLM] Add headers and API for optimized kernels for CEVA-DSP BX1 and SP500

### PR DESCRIPTION
**Pull request by yair-ehrenwald**
Added common headers and a scratch allocation file for CEVA platforms.

These are linked against an internal CEVA static lib containing all the optimized code. Since the library is not open source (or part of a third_party_download), we are currently adding the headers directly to the TFLM repository.

This is the same reason why these headers match the CEVA API and style (as opposed to the Google C++ style) and are not reviewed beyond looking at license compliance.

In the future, there might be value in pulling in the library and headers as a combined third_party_download. If that turns out to be useful, we would remove these headers from the TFLM repository.

Relevant github issue: https://github.com/tensorflow/tensorflow/issues/45607

### CODE REVIEW

File: tensorflow/lite/micro/kernels/ceva/ceva_common.cc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/micro/kernels/ceva/ceva_tflm_lib.h"
+#define CEVA_TFLM_KERNELS_SCRATCH_SIZE_VAL_DEF 32768
+int32_t CEVA_TFLM_KERNELS_SCRATCH_SIZE_VAL = 32768;
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/ceva/ceva_common.cc:18]: what is the reason for both the #define and int32_t?

File: tensorflow/lite/micro/kernels/ceva/ceva_common.cc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#ifndef WIN32
+__attribute__((section(".MODEL_DATA")))
+#endif
+int32_t CEVA_TFLM_KERNELS_SCRATCH[CEVA_TFLM_KERNELS_SCRATCH_SIZE_VAL_DEF];
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/ceva/ceva_common.cc:22]: if possible change to:
```cc
int32_t ceva_tflm_kernels_scratch[CEVA_TFLM_KERNELS_SCRATCH_SIZE_VAL_DEF];
```

https://google.github.io/styleguide/cppguide.html#Variable_Names

And I'm a bit confused how this array will be accessed unless it is also declared in a header?

**Review by advaitjain — COMMENTED** [review]
small clarifying comments and then we can merge this PR.

**Comment by advaitjain** [comment]
Even though we're ok with the headers not following the google-style, we will still need to clang-format the files.

File: tensorflow/lite/micro/kernels/ceva/ceva_common.cc (hunk 1 above)
Comment (yair-ehrenwald) [reply tensorflow/lite/micro/kernels/ceva/ceva_common.cc:18]: To conform with the API used in the functions, they check by comparing to the in32_t - accessed by defining both the size value and the buffer itself as "extern" (which also answers the question bellow).
There is however no reason not to define the variable by the #define, I'll fix that.

File: tensorflow/lite/micro/kernels/ceva/ceva_common.cc (hunk 2 above)
Comment (yair-ehrenwald) [reply tensorflow/lite/micro/kernels/ceva/ceva_common.cc:22]: I clang-formatted the files. I'd rather not change the name as it is used throughout our code...

**Review by advaitjain — APPROVED** [review]

File: tensorflow/lite/micro/kernels/ceva/ceva_common.cc (hunk 2 above)
Comment (advaitjain) [reply tensorflow/lite/micro/kernels/ceva/ceva_common.cc:22]: makes sense.

**Pull request merged**